Translation/Extractor/FileExtractor.php

```diff
@@ -53,8 +56,13 @@ public function __construct(\Twig_Environment $twig, LoggerInterface $logger, ar
         $this->twig = $twig;
         $this->logger = $logger;
         $this->visitors = $visitors;
-        $lexer = new \PHPParser_Lexer();
-        $this->phpParser = new \PHPParser_Parser($lexer);
+        $lexer = new Lexer();
+        if(class_exists('PhpParser\ParserFactory')) {
+            $factory = new ParserFactory();
+            $this->phpParser = $factory->create(ParserFactory::PREFER_PHP7,$lexer);
+        } else {
+            $this->phpParser = new \PHPParser_Parser($lexer);
+        }
 
```

Comment:
So I'm not going to implement this suggestion here for a couple of reasons.
* Most of the instances of these lines of code are within the tests themselves (which means I guess we could create a base class to do the initialization)
* The factory class would have to handle all the options you can pass to the actual ParserFactory which don't apply to the parser 1.4 branch, meaning we would then also need to duplicate their constants since this isn't an interface we can abstract. 
* Because of the two and the minor improvements I think we just not worry about this bit.

Thoughts?